File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
# 모든달은 28일까지있음

def solution(today, terms, privacies):
    answer = []
    dic = {}
    for i in terms:
        list = i.split()
        dic[list[0]] = int(list[1])
    
    # list[0] 년
    # list[1] 월
    # list[2] 일
    # list[3] 기준
    
    today_list = today.split(".")
    today_sum = int(today_list[0])*12*28 + int(today_list[1])*28 + int(today_list[2])
    cnt = 1
    for i in privacies:
        replace_i = i.replace(' ','.')
        list = replace_i.split(".")    
        sum = int(list[0])*12*28 + int(list[1])*28 + int(list[2])
        
        if today_sum - sum >= int(dic[list[3]]) * 28:
            answer.append(cnt)
        cnt += 1
    
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    const answer = [];
    
    const endDateObject = {}
    terms.forEach((endDate)=> {
        const date = endDate.split(" ")
        endDateObject[date[0]] = parseInt(date[1])
    })
    

    
    const todayList = today.split(".").map(item => parseInt(item))
    
    privacies.forEach((createdAndEndDate,index)=> {
        const items = createdAndEndDate.split(" ")
        const date = items[0].split('.').map(item => parseInt(item))
        
        // 유효기한
        const expireMonth = endDateObject[items[1]]
        
        date[1] += expireMonth
    
        if (date[1] > 12) {
            date[0] += Math.floor((date[1] - 1) / 12);
            date[1] = (date[1] - 1) % 12 + 1;
        }

        if (
            date[0] < todayList[0] || 
            (date[0] === todayList[0] && date[1] < todayList[1]) ||
            (date[0] === todayList[0] && date[1] === todayList[1] && date[2] <= todayList[2])
        ) {
            answer.push(index + 1);
        }
    })
    
    return answer;
}